mirrord/agent/src/steal/connection.rs

```diff
@@ -649,14 +683,26 @@ impl TcpConnectionStealer {
                         ConnectionMessageIn::Unsubscribed { client_id },
                     )
                     .await;
+
+                STEAL_CONNECTION_SUBSCRIPTION.dec();
             }
 
             Command::PortSubscribe(port_steal) => {
-                self.port_subscribe(client_id, port_steal).await?
+                if self.port_subscribe(client_id, port_steal).await? {
+                    STEAL_FILTERED_PORT_SUBSCRIPTION.inc();
+                } else {
+                    STEAL_UNFILTERED_PORT_SUBSCRIPTION.inc();
+                }
             }
 
             Command::PortUnsubscribe(port) => {
-                self.port_subscriptions.remove(client_id, port).await?;
+                if let Some(filtered) = self.port_subscriptions.remove(client_id, port).await? {
+                    if filtered {
+                        STEAL_FILTERED_PORT_SUBSCRIPTION.inc();
+                    } else {
+                        STEAL_UNFILTERED_PORT_SUBSCRIPTION.inc();
+                    }
+                }
```

Comment:
I'd move updating port subscription metrics to `PortSubscriptions`. You won't have to return the bools from its methods